modules/core/src/main/java/org/apache/ignite/internal/processors/resource/GridResourceIoc.java

```diff
@@ -574,6 +574,16 @@ public boolean inject(Object target,
             ResourceAnnotation.IGNITE_INSTANCE,
             ResourceAnnotation.LOGGER,
             ResourceAnnotation.SERVICE
+        ),
+
+        /**
+         * Same as {@link #GENERIC} excluding {@link ResourceAnnotation#LOGGER}.
+         */
+        GENERIC_CLEANABLE(
```

Comment:
Can we just exclude `LOGGER` from `GENERIC` set instead of the creation of a new set?
New set is error-prone.
Some contributor can forget to add new `ReourceAnnotation` to it.
This will lead to hard to find issues.

Reply:
Is it right that you suggest to add a new AnnotationSet.LOGGER with ResourceAnnotation.LOGGER only and use it at the points where AnnotationSet.GENERIC is injecting but don't use one at cleanup?

Reply:
I think we can just use existing LOGGER annotation but don't clean up it.

Reply:
Thanks for review, code was updated according your recommendation.